octobot_trading/personal_data/portfolios/portfolio_value_holder.py

```diff
@@ -223,27 +224,37 @@ def _try_get_value_of_currency(self, currency, quantity, raise_error):
         It will try to create the symbol that fit with the exchange logic.
         :return: the value found of this currency quantity, if not found returns 0.
         """
-        symbol = symbol_util.merge_currencies(currency, self.portfolio_manager.reference_market)
-        reversed_symbol = symbol_util.merge_currencies(self.portfolio_manager.reference_market, currency)
-
         try:
-            if self.portfolio_manager.exchange_manager.symbol_exists(symbol):
-                return self.last_prices_by_trading_pair[symbol] * quantity
-
-            if self.portfolio_manager.exchange_manager.symbol_exists(reversed_symbol) and \
-                    self.last_prices_by_trading_pair[reversed_symbol] != constants.ZERO:
-                return quantity / self.last_prices_by_trading_pair[reversed_symbol]
-
-            if currency not in self.missing_currency_data_in_exchange:
+            return self.convert_currency_value_using_last_prices(quantity, currency,
+                                                                 self.portfolio_manager.reference_market)
+        except errors.MissingPriceDataError as missing_data_exception:
+            symbol = symbol_util.merge_currencies(currency, self.portfolio_manager.reference_market)
+            reversed_symbol = symbol_util.merge_currencies(self.portfolio_manager.reference_market, currency)
+            if not any(self.portfolio_manager.exchange_manager.symbol_exists(s) for s in (symbol, reversed_symbol)) \
+               and currency not in self.missing_currency_data_in_exchange:
                 self._inform_no_matching_symbol(currency)
                 self.missing_currency_data_in_exchange.add(currency)
-        except KeyError as missing_data_exception:
             if not self.portfolio_manager.exchange_manager.is_backtesting:
                 self._try_to_ask_ticker_missing_symbol_data(currency, symbol, reversed_symbol)
                 if raise_error:
                     raise missing_data_exception
         return constants.ZERO
 
+    def convert_currency_value_using_last_prices(self, quantity, current_currency, target_currency):
+        try:
+            symbol = symbol_util.merge_currencies(current_currency, target_currency)
+            if self.last_prices_by_trading_pair[symbol] is not constants.ZERO:
+                return quantity * self.last_prices_by_trading_pair[symbol]
+        except KeyError:
+            pass
+        try:
+            reversed_symbol = symbol_util.merge_currencies(target_currency, current_currency)
+            if self.last_prices_by_trading_pair[reversed_symbol] is not constants.ZERO:
```

Comment:
I think we should catch DivisionError or OperationError instead of using a if here